ubuntu-advantage

```diff
@@ -141,6 +154,109 @@ disable_esm() {
+    fips_hmacs="openssh-client-hmac openssh-server-hmac libssl1.0.0-hmac \
+                linux-fips strongswan-hmac"
+    for pkg in $fips_hmacs; do
+        dpkg-query -W $pkg > /dev/null 2>&1
+        if [ $? -eq 0 ]; then
+            echo "ERROR: Looks like FIPS already installed."
+            return 1
+        fi
+    done
+
+    # sanity check
+    case "$ARCH" in
+        s390x|x86_64|ppc64le)
+            ;;
+        *)
+            echo "ERROR: $ARCH is not fips supported."
+            return 1
+            ;;
+    esac
+    return 0
+}
+
+configure_fips() {
+    # if /boot has its own partition, then get the bootdevice
+    # Note: /boot/efi  does not count
+    bootdev=$(awk '!/^\s*#/ && $2 ~ /^\/boot\/?$/ { print $1 }' $FSTAB)
+    fips_params="fips=1"
+    if [ -n "$bootdev" ]; then
+        fips_params="$fips_params bootdev=$bootdev"
+    fi
+
+    if [ $ARCH = 's390x' ]; then
+        sed -i -e 's/^parameters\s*=.*/& '"$fips_params"'/' "$FIPS_BOOT_CFG"
+        echo -n 'Updating zipl to enable fips ...'
+        check_result zipl
+    else
+        if ! test -d $FIPS_BOOT_CFG_DIR; then
+            mkdir $FIPS_BOOT_CFG_DIR > /dev/null 2>&1
+            if [ $? -ne 0 ]; then
+                echo "Failed to make directory, $FIPS_BOOT_CFG_DIR."
+                return 1
+            fi
+        fi
+        cat > "$FIPS_BOOT_CFG" <<EOF
+GRUB_CMDLINE_LINUX_DEFAULT="\$GRUB_CMDLINE_LINUX_DEFAULT $fips_params"
+EOF
+        echo -n 'Updating grub to enable fips ...'
+        check_result update-grub
+    fi
+}
+
+write_fips_list_file() {
+    cat > "$FIPS_REPO_LIST" <<EOF
+deb https://${1}@${FIPS_REPO_URL}/ubuntu ${SERIES} main
+# deb-src https://${1}@${FIPS_REPO_URL}/ubuntu ${SERIES} main
+EOF
+}
+
+enable_fips() {
+    fips_prep_check || exit 6
+
+    cp "${KEYRINGS_DIR}/${FIPS_REPO_KEY_FILE}" "$APT_KEYS_DIR"
+    write_fips_list_file "$1"
+
+    if [ ! -f "$APT_METHOD_HTTPS" ]; then
+        echo -n 'Installing missing dependency apt-transport-https... '
+        check_result apt_get install apt-transport-https
+    fi
+    if [ ! -f "$CA_CERTIFICATES" ]; then
+        echo -n 'Installing missing dependency ca-certificates... '
+        check_result apt_get install ca-certificates
+    fi
+    echo -n 'Running apt-get update... '
+    check_result apt_get update
+    echo 'Ubuntu FIPS PPA repository enabled.'
+
+    # install all the fips packages
+    echo -n 'Installing FIPS packages (this may take a while)...'
+    check_result apt_get install openssh-client openssh-client-hmac \
+                       openssh-server openssh-server-hmac openssl libssl1.0.0 \
+                       libssl1.0.0-hmac fips-initramfs linux-fips \
+                       strongswan strongswan-hmac
+
+    echo "Configuring FIPS..."
+    configure_fips
+    echo "Successfully configured FIPS. PLEASE REBOOT to complete FIPS enablement."
+}
+
+fips_enabled_check() {
+    if [ -f "$FIPS_ENABLED_FILE" ]; then
+                cat $FIPS_ENABLED_FILE
```

Comment:
Why do you need to cat the file?

Reply:
/proc/sys/crypto/fips_enabled contains either a "0" (fips mode is disabled) or a "1" (fips mode is enabled). Cat the file will tell me which one it is...